celery/backends/base.py

```diff
@@ -428,9 +430,13 @@ def iter_native(self, result, timeout=None, interval=0.5, no_ack=True,
         )
 
     def wait_for_pending(self, result, timeout=None, interval=0.5,
-                         no_ack=True, on_interval=None, callback=None,
-                         propagate=True):
+                         no_ack=True, on_message=None, on_interval=None,
+                         callback=None, propagate=True):
         self._ensure_not_eager()
+        if on_message is not None:
```

Comment:
Wonder if this condition can cause following issue? Mq backed is not supported for on_message ? @xBeAsTx @auvipy 
http://stackoverflow.com/questions/40784593/celery-raise-improperlyconfigured-exception